src/main/java/com/aws/greengrass/mqttclient/spool/Spool.java

```diff
@@ -81,11 +90,43 @@ private void setSpoolerConfigFromDeviceConfig(Topics topics) {
     private CloudMessageSpool setupSpooler() {
         if (config.getStorageType() == SpoolerStorageType.Memory) {
             return new InMemorySpool();
+        } else if (config.getStorageType() == SpoolerStorageType.FileSystem) {
```

Comment:
I don't think this storage type should be called filesystem. This is actually more of a generic solution, since it will find any plugin and can thus be implemented in any way. Suggest changing FileSystem -> Plugin.

Reply:
I think the naming of this was done before me, but I can change it